feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,537 @@
+from typing import List, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.dataframe_checks import (
+    _check_contains_inf,
+    _check_contains_na,
+    _is_dataframe,
+)
+from feature_engine.discretisation import (
+    EqualFrequencyDiscretiser,
+    EqualWidthDiscretiser,
+)
+from feature_engine.selection.base_selector import BaseSelector
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+Variables = Union[None, int, str, List[Union[str, int]]]
+
+
+class DropHighPSIFeatures(BaseSelector):
+    """
+    DropHighPSIFeatures drops features with a Population Stability Index (PSI) above
+    a given threshold. The PSI of a feature is an indication of the shift in its
+    distribution; a feature with high PSI might therefore be seen as instable.
+
+    In fields like Credit Risk Modelling, the elimination of features with high PSI
+    is frequent and usually required by the Regulator.
+
+    The calculation of the PSI requires to compare two distributions.
+    In DropHighPSIFeatures the approach implemented is the following:
+
+    - The dataframe is split in two parts and the PSI calculations are performed by
+    comparing the distribution of the features in the two parts.
+
+    - The user defines the columns of the dataframe according to which the dataframe
+    will de split. The column can contain numbers, dates and strings.
+
+    - The user defines a split fraction. This is used to define the sizes of the two
```

Comment:
this would only be used if the split column is numerical? If dates, then it does not apply, right?

Reply:
In the current case, I made it generic so it would be used in all cases. This would say for example split my time scale in two equal parts (or in 75%, 25%). This "replace" them a cut-off date.